test/src/in_language_tests/test_programs/b512_inline_tests/src/main.sw

```diff
@@ -0,0 +1,284 @@
+        0x0000000000000000000000000000000000000000000000000000000000000000,
+        0x0000000000000000000000000000000000000000000000000000000000000000,
+    );
+    let b512_1: B512 = b256_tuple_1.into();
+    assert(
+        b512_1
+            .bits()[0] == 0x0000000000000000000000000000000000000000000000000000000000000000 && b512_1
+            .bits()[1] == 0x0000000000000000000000000000000000000000000000000000000000000000,
+    );
+
+    let b256_tuple_2: (b256, b256) = (
+        0x0000000000000000000000000000000000000000000000000000000000000001,
+        0x0000000000000000000000000000000000000000000000000000000000000001,
+    );
+    let b512_2: B512 = b256_tuple_2.into();
+    assert(
+        b512_2
+            .bits()[0] == 0x0000000000000000000000000000000000000000000000000000000000000001 && b512_2
+            .bits()[1] == 0x0000000000000000000000000000000000000000000000000000000000000001,
+    );
+
+    let b256_tuple_3: (b256, b256) = (
+        0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+        0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+    );
+    let b512_3: B512 = b256_tuple_3.into();
+    assert(
+        b512_3
+            .bits()[0] == 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF && b512_3
+            .bits()[1] == 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+    );
+}
+
+#[test]
+fn b512_b256_from() {
+    // Glob operator needed for From<B512> for (b256, b256)
+    use std::b512::*;
+
+    let b512_1 = B512::from((
+        0x0000000000000000000000000000000000000000000000000000000000000000,
+        0x0000000000000000000000000000000000000000000000000000000000000000,
+    ));
+    let b256_tuple_1: (b256, b256) = b512_1.into();
+    assert(
+        b256_tuple_1.0 == 0x0000000000000000000000000000000000000000000000000000000000000000 && b256_tuple_1.1 == 0x0000000000000000000000000000000000000000000000000000000000000000,
+    );
+
+    let b512_2 = B512::from((
+        0x0000000000000000000000000000000000000000000000000000000000000001,
+        0x0000000000000000000000000000000000000000000000000000000000000001,
+    ));
+    let b256_tuple_2: (b256, b256) = b512_2.into();
+    assert(
+        b256_tuple_2.0 == 0x0000000000000000000000000000000000000000000000000000000000000001 && b256_tuple_2.1 == 0x0000000000000000000000000000000000000000000000000000000000000001,
+    );
+
+    let b512_3 = B512::from((
+        0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+        0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+    ));
+    let b256_tuple_3: (b256, b256) = b512_3.into();
+    assert(
+        b256_tuple_3.0 == 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF && b256_tuple_3.1 == 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF,
+    );
+}
+
+#[test]
+fn b512_b256_into() {
+    // Glob operator needed for From<B512> for (b256, b256)
```

Comment:
Todo for when this is fixed? (And all other instances of glob) other than that looks good

Reply:
When importing `B512` with `use std::b512::B512;`, this does not import implementation on the `b256` type i.e. `impl From<B512> for b256` . I do not believe there are any intentions to include associated implementations such as this `From` implementation when importing a data structure. cc @IGI-111

Reply:
There is no short term fix coming for this. We need proper impl coherence which is a long term thing.